README.md

```diff
@@ -139,6 +141,162 @@ Remember to update the file and any new experiments added to the `filesystem` di
+It reuses the same SolrCloud service described earlier, and a Postgres container with minor variations: it doesn’t use
+volumes to ensure the database is clean before running any tests, and its name (and the dependency expressed in
+`docker-compose-gradle.yml`) has been changed to `gxa-postgres-test`; the reason is to avoid using `gxa-postgres` by
+mistake and wiping full tables when cleaning fixtures... such an unfortunate accident is known to have happened.
+
+Depending on your OS and Docker settings you might be able to run Gradle from your host machine without a container,
+and access Solr, ZooKeeper and Postgres via mapped ports on `localhost`. We know this is possible in Linux, but we’ve
+found it to be problematic in macOS. It’s probably due to the way DNS in Docker Compose works (i.e., ZooKeeper resolves
+`localhost` to an unknown IP address). As they say, networking is hard. YMMV.
+
+### Before you start
+Check with `docker ps` and `docker container ls -a` that no services used during tests are running or stopped,
+respectively. These are `gxa-solrcloud-1`, `gxa-solrcloud-2`, `gxa-zk-1`, `gxa-zk-2`, `gxa-zk-3`,
+`gxa-postgres-test`, `gxa-flyway-test` and `gxa-gradle`. We want to start with a clean application context every
+time we execute the test task. Here are two useful commands:
+```bash
+docker stop scxa-solrcloud-1 scxa-solrcloud-2 scxa-zk-1 scxa-zk-2 scxa-zk-3 scxa-postgres-test scxa-flyway-test scxa-gradle
+```
+
+```bash
+docker container prune
+```
+
+### Running tests
+As mentioned before, `docker-compose-gradle.yml` runs the Gradle `test` task and it depends on all the necessary
+services to run unit tests, integration tests and end-to-end tests. It splits the job in the following six phases:
+1. Clean the build directory
+2. Compile the test classes
+3. Run unit tests
+4. Run integration tests
+5. Run end-to-end tests
+6. Generate JaCoCo reports
+
+Bring it up like this (the Postgres variables can take any values, remember that the container will be removed):
+```bash
+ATLAS_DATA_PATH=/path/to/your/scxa/data \
+POSTGRES_HOST=gxa-postgres-test \
+POSTGRES_DB=gxpgxatest \
+POSTGRES_USER=gxa \
+POSTGRES_PASSWORD=gxa \
+docker-compose \
+-f docker-compose-postgres-test.yml \
+-f docker-compose-solrcloud.yml \
+-f docker-compose-gradle.yml \
+up
+```
+
+You will eventually see these log messages:
+```
+gxa-gradle         | BUILD SUCCESSFUL in 13s
+gxa-gradle         | 3 actionable tasks: 1 executed, 2 up-to-date
+gxa-gradle exited with code 0
+```
+
+Press Ctrl+C to stop the container and clean any leftovers:
+```bash
+docker stop gxa-solrcloud-1 gxa-solrcloud-2 gxa-zk-1 gxa-zk-2 gxa-zk-3 gxa-postgres-test gxa-flyway-test &&
+docker rm gxa-solrcloud-1 gxa-solrcloud-2 gxa-zk-1 gxa-zk-2 gxa-zk-3 gxa-postgres-test gxa-flyway-test gxa-gradle
+```
+Another way of removing all stopped containers in one line you can use following statement:
+
+```bash
+docker container prune
+```
```

Comment:
I strongly advise against this. You might have other stopped containers unrelated to the project (I usually do) that I don’t want to remove.